DataFormats/ParticleFlowCandidate/interface/PFCandidate.h

```diff
@@ -429,6 +405,41 @@ namespace reco {
     /// set the fraction of hcal energy as function of depth (index 0..6 for depth 1..7)
     void setHcalDepthEnergyFractions(const std::array<float, 7>& fracs) { hcalDepthEnergyFractions_ = fracs; }
 
+    const math::XYZPoint& vertexLegacy(PFVertexType vertexType) const {
+      std::cout << "vertexLegacy called with " << vertexType << std::endl;
+      switch (vertexType) {
+        case kCandVertex:
+          return LeafCandidate::vertex();
+          break;
+        case kTrkVertex:
+          return trackRef()->vertex();
```

Comment:
Oh! Thanks for the investigation, this would have taken me some time to figure out, but I guess it makes sense. So it looks like we have to rethink our approach.